### PR TITLE
52 - Add special modal with data storage disclaimer

### PR DESCRIPTION
**Pull request by rivasvict**
## Description

Added a disclaimer as a higher order component that can be easily removed.
image
image

## How to test:

In order to test this change, you can go to https://deploy-preview-64--vortracker.netlify.app/ and try:
1. Without clicking the checkbox, press "I Aknowledge" and then refresh, note that the modal will appear again.
2. Select the checkbox and then press "I Aknowledge" and then refresh, note that the modal will not appear again.
3. Use `localStorage.removeItem('everShowDataDisclaimer')` to clear the value and try 1 and 2 from this list again.

You can also see the acceptance criteria in the original issue https://github.com/rivasvict/react-expenses-manager/issues/52

### CODE REVIEW

**Review by ricardo-vargas — COMMENTED** [review]
I might be wrong but I want to ask, Why adding a simple modal to state? couldn't we just directly save to a cookie?

File: src/redux/userManager/actoionCreators.js
Comment (ricardo-vargas) [inline src/redux/userManager/actoionCreators.js:1]: file name could be confusing, is it a typo?

Comment (ricardo-vargas) [inline src/redux/userManager/actoionCreators.js:1]: file name could be confusing, is it a typo?

File: src/redux/userManager/reducer.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ const getDefaultState = () => ({
     lastName: null,
     isLoading: true,
   },
+  everShowDataDisclaimer: true,
```
Comment (ricardo-vargas) [inline src/redux/userManager/reducer.js:18]: Do we need this property in the reducer?

File: src/App.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import Dashboard from "./components/Dashboard/Dashboard";
 import { Provider, connect } from "react-redux";
 import { setUser } from "./redux/userManager/actoionCreators";
```
Comment (ricardo-vargas) [inline src/App.js:11]: same here, is this a typo?

File: src/components/Dashboard/WithDataDisclaimer/components/DataDisclaimer/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from "react";
+import { Button, Col, Container, Form, Modal } from "react-bootstrap";
+import { BOOLEAN_ENUM } from "../../../../../constants";
```
Comment (ricardo-vargas) [inline src/components/Dashboard/WithDataDisclaimer/components/DataDisclaimer/index.js:3]: just as a suggestion, something that can be done in the future is to add a path alias to make this more readable

**Review by ricardo-vargas — APPROVED** [review]

File: src/components/Dashboard/WithDataDisclaimer/components/DataDisclaimer/index.js (hunk 3 above)
Comment (rivasvict) [reply src/components/Dashboard/WithDataDisclaimer/components/DataDisclaimer/index.js:3]: Great suggestion, thank you so so much, appreciate it.

File: src/redux/userManager/reducer.js (hunk 1 above)
Comment (rivasvict) [reply src/redux/userManager/reducer.js:18]: Good catch

File: src/redux/userManager/actoionCreators.js
Comment (rivasvict) [reply src/redux/userManager/actoionCreators.js:1]: Yes, great catch

File: src/components/Dashboard/WithDataDisclaimer/components/DataDisclaimer/index.js (hunk 3 above)
Comment (rivasvict) [reply src/components/Dashboard/WithDataDisclaimer/components/DataDisclaimer/index.js:3]: I'll add it in another ticket https://github.com/rivasvict/react-expenses-manager/issues/66

**Pull request merged**